design-editor/src/pane/style-manager.js

```diff
@@ -58,28 +58,97 @@ class StyleManager {
         eventEmitter.on(EVENTS.ChangeElementStyle, this._onChangeElementStyle.bind(this));
     }
 
-    /**
-     * Change element style callback
-     * @param {string} template
-     * @private
-     */
-    _onChangeElementStyle(template) {
-        console.log('style-manager:_onChangeElementStyle');
+	/**
+	 * Change element style callback
+	 * @param {string | object} newStyleDefinition
+	 * @private
+	 */
+	_onChangeElementStyle(newStyleDefinition) {
+		// eslint-disable-next-line no-console
+		console.log('style-manager:_onChangeElementStyle');
+		const id = elementSelector.getSelectedElementId(),
+			designEditor = appManager.getActiveDesignEditor(),
+			$element = designEditor && $(designEditor._getElementById(id));
+		let template;
 
-        var id = elementSelector.getSelectedElementId(),
-            designEditor = appManager.getActiveDesignEditor(),
-            $element = designEditor && $(designEditor._getElementById(id));
+		this._designEditor = designEditor;
+		if (designEditor.getUIInfo($element).package.name === 'listview') {
+			this._getModel($element.children(), 'listview');
+		} else {
+			this._getModel($element, designEditor.getUIInfo($element).package.name);
+		}
+		if (typeof newStyleDefinition === 'string') {
+			template = newStyleDefinition;
+			this._replaceStyle($element, template);
+		} else {
+			this._applyStyle($element, newStyleDefinition);
+		}
+	}
 
-        this._designEditor = designEditor;
+	/**
+	 * Applies defined style for element
+	 * @param {jQuery} $element
+	 * @param {Object} newStyleDefinition
+	 */
+	_applyStyle($element, newStyleDefinition) {
+		const id = $element.attr('data-id'),
+			model = this._designEditor.getModel(),
+			element = model.getElement(id),
+			component = element && element.component,
+			currentClassList = element.classList,
+			newClassList = newStyleDefinition.classList || [],
+			newAttributes = newStyleDefinition.attributes || {},
+			newStyles = newStyleDefinition.style || {},
+			classesSet = new Set(); // classes used across widget available templates
```

Comment:
Other way without Set
```
classes = {}
newStyleDefinition.template.classList.forEach(c => classes[c] = true);
currentClassList = currentClassList.filter(c => !classes[c]);
currentClassList = currentClassList.concat(newClassList);
```

Reply:
I think we can leave Set implementation.
Set operations are faster than operations based on Arrays or Objects (in this case it don't think it would have huge impact, but I would leave the current implementation).

Benchmark of adding elements and checking if collections has an key: https://github.com/anvaka/set-vs-object